internal/app/upstream/multiplexer.go

```diff
@@ -0,0 +1,55 @@
+package multiplexer
+
+import (
+	"context"
+
+	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"google.golang.org/grpc"
+)
+
+// Multiplexer handles the requests and responses from the origin server.
+// The multiplexer handles each xds request on a separate stream,
+// e.g. 2 different cds requests happen on 2 separate streams.
+// It is the caller's responsibility to make sure there is one instance of Multipler per unique xds request.
```

Comment:
typo "Multipler" ?

Reply:
I'm still a little confused what this interface is for. What is it multiplexing exactly if there is one required for each unique xDS request and the caller is responsible for figuring out uniqueness? Is this just an `XdsClient` interface?

Reply:
That is true and I've now renamed it to XdsClient.